starport/services/network/networkchain/prepare.go

```diff
@@ -205,30 +206,64 @@ func (c Chain) applyGenesisValidators(ctx context.Context, genesisVals []network
 // updateConfigFromGenesisValidators adds the peer addresses into the config.toml of the chain
 func (c Chain) updateConfigFromGenesisValidators(genesisVals []networktypes.GenesisValidator) error {
 	var p2pAddresses []string
-	for _, val := range genesisVals {
-		p2pAddresses = append(p2pAddresses, val.Peer.GetTcpAddress())
+	var tunnelAddresses []TunneledPeer
+	for i, val := range genesisVals {
+		if val.Peer.GetTcpAddress() != "" {
+			p2pAddresses = append(p2pAddresses, val.Peer.GetTcpAddress())
+		} else {
+			tunnel := val.Peer.GetHttpTunnel()
+			addressParts := strings.Split(tunnel.Address, "@")
```

Comment:
I think we can use the `cosmosutil` validation here: `cosmosutil.VerifyPeerFormat(peer)`